pkg/testcase/secretsencrypt.go

```diff
@@ -0,0 +1,137 @@
+package testcase
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/rancher/distros-test-framework/shared"
+
+	. "github.com/onsi/gomega"
+)
+
+func TestSecretsEncrypt() {
+	etcdNodes, errGetEtcd := shared.GetNodesByRoles("etcd")
+	Expect(etcdNodes).NotTo(BeEmpty())
+	Expect(errGetEtcd).NotTo(HaveOccurred(), "error getting etcd nodes")
+
+	cpNodes, errGetCP := shared.GetNodesByRoles("control-plane")
+	Expect(cpNodes).NotTo(BeEmpty())
+	Expect(errGetCP).NotTo(HaveOccurred(), "error getting control plane nodes")
+
+	product, err := shared.Product()
+	Expect(err).NotTo(HaveOccurred(), "error getting product from config")
+
+	ips := getNodeIps(etcdNodes, cpNodes)
+
+	errSecret := shared.CreateSecret("secret1", "default")
+	Expect(errSecret).NotTo(HaveOccurred(), "error creating secret")
+
+	secretsEncryptOps("prepare", product, cpNodes[0].ExternalIP, ips)
+	secretsEncryptOps("rotate", product, cpNodes[0].ExternalIP, ips)
+	secretsEncryptOps("reencrypt", product, cpNodes[0].ExternalIP, ips)
+	secretsEncryptOps("rotate-keys", product, cpNodes[0].ExternalIP, ips)
+}
+
+func secretsEncryptOps(action, product, cpIp string, ips []string) {
```

Comment:
can we return error please ? since has some possible areas that could go wrong, would be good to send the error above